### PR TITLE
Track user profile views

### PR DESCRIPTION
**Pull request by tgxworld**
Meta: https://meta.discourse.org/t/is-there-a-way-to-track-profile-page-views/32385/21

### CODE REVIEW

**Comment by ZogStriP** [comment]
Seems like it needs a rebase

**Comment by eviltrout** [comment]
PR looks excellent. Would love to merge it when it's ready.

File: app/models/user_profile_view.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class UserProfileView < ActiveRecord::Base
+  validates_presence_of :user_profile_id, :ip_address, :viewed_at
+
+  belongs_to :user_profile
+
+  def self.add(user_profile_id, ip, user_id=nil, at=nil, skip_redis=false)
+    at ||= Time.zone.now
+    redis_key = "user-profile-view:#{user_profile_id}:#{at.to_date}"
```
Comment (tgxworld) [inline app/models/user_profile_view.rb]: @coding-horror @eviltrout I was speaking with @SamSaffron  about this. Right now, we're resetting the "view counter" daily apart from expiring the key every 8 hours. That means I can register two topic view counts by viewing the topic at 11pm and then 12am. Is this the intended behaviour?

**Comment by ZogStriP** [comment]
Thanks :+1:

**Pull request merged**